src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -183,6 +206,230 @@ public void initAverRestCommunicator() throws Exception {
 	 */
 	@Override
 	public void controlProperty(ControllableProperty controllableProperty) {
+		String property = controllableProperty.getProperty();
+		String value = String.valueOf(controllableProperty.getValue());
+		controlOperationsLock.lock();
+
+		if (this.logger.isDebugEnabled()) {
+			this.logger.debug("controlProperty property " + property);
+			this.logger.debug("controlProperty value " + value);
+		}
+
+		try {
+			String[] splitProperty = property.split(String.valueOf(HASH));
+			Command command = Command.getByName(splitProperty[0]);
+
+			switch (command) {
+				case POWER: {
+					if (value.equals(SWITCH_STATUS_ON)) {
+						powerOn();
+					} else if (value.equals(SWITCH_STATUS_OFF)) {
+						powerOff();
+					}
+					break;
+				}
+				case ZOOM: {
+					if (splitProperty[1].equals(ZoomControl.TELE.getName())) {
+						zoomTele();
+					} else if (splitProperty[1].equals(ZoomControl.WIDE.getName())) {
+						zoomWide();
+					}
+					break;
+				}
+				case FOCUS: {
+					if (splitProperty[1].equals(Command.FOCUS_MODE.getName())) {
+						if (value.equals(SWITCH_STATUS_ON)) {
+							manualFocus();
+						} else if (value.equals(SWITCH_STATUS_OFF)) {
+							autoFocus();
+						}
+						break;
+					}
+
+					if (splitProperty[1].equals(Command.FOCUS_ONE_PUSH.getName())) {
+						onePushFocus();
+						break;
+					}
+
+					if (splitProperty[1].equals(FocusControl.FAR.getName())) {
+						focusFar();
+					} else if (splitProperty[1].equals(FocusControl.NEAR.getName())) {
+						focusNear();
+					}
+					break;
+				}
+				case EXPOSURE: {
```

Comment:
EXPOSURE, IMAGE_PROCESS and PAN_TILT_DRIVE are the long processing part. I think we should separate these parts into separate methods